tests/js/realm-tests.js

```diff
@@ -1288,42 +1282,48 @@ module.exports = {
 
     // FIXME: We need to test adding a property also calls the listener
     testSchemaUpdatesNewClass: function() {
+        let realm1 = new Realm({ _cache: false });
+        TestCase.assertTrue(realm1.empty);
+        TestCase.assertEqual(realm1.schema.length, 0);  // empty schema
+
+        const schema = [{
+            name: 'TestObject',
+            properties: {
+                prop0: 'string',
+            }
+        }];
+
         return new Promise((resolve, reject) => {
-            let called = false;
-            let realm1 = new Realm({ _cache: false });
-            TestCase.assertTrue(realm1.empty);
-            TestCase.assertEqual(realm1.schema.length, 0);  // empty schema
             realm1.addListener('schema', (realm, event, schema) => {
-                TestCase.assertEqual(event, 'schema');
-                TestCase.assertEqual(schema.length, 1);
-                TestCase.assertEqual(realm.schema.length, 1);
-                TestCase.assertEqual(schema[0].name, 'TestObject');
-                TestCase.assertEqual(realm1.schema.length, 1);
-                TestCase.assertEqual(realm.schema[0].name, 'TestObject');
-                called = true;
-            });
-
-            const schema = [{
-                name: 'TestObject',
-                properties: {
-                    prop0: 'string',
+                try {
+                    TestCase.assertEqual(event, 'schema');
+                    if (schema.length == 0) {
```

Comment:
How can the length be 0...doesn't that indicate a problem with the test somehow?

Reply:
Yes, that's leftover code from tracking down a problem that I forgot to remove.